rosbag2_performance/rosbag2_performance_writer_benchmarking/include/message_queue.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020, Robotec.ai sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_PERFORMAMCE_WRITER_BENCHMARKING__MESSAGE_QUEUE_HPP_
+#define ROSBAG2_PERFORMAMCE_WRITER_BENCHMARKING__MESSAGE_QUEUE_HPP_
+
+#include <string>
+#include <queue>
+#include <utility>
+#include <mutex>
+#include <atomic>
+#include <iostream>
+#include <memory>
+
+#include "std_msgs/msg/byte_multi_array.hpp"
+
+template<typename T>
+class MessageQueue
+{
+public:
+  MessageQueue(int maxSize, std::string topicName)
+  : mMaxSize(maxSize), mTopicName(topicName), mUnsuccessfulInsertCount(0) {}
+
+  void push(T elem)
```

Comment:
if we deal with shared_pointers exclusively, then I would template it accordingly. Either take a `std::shared_ptr<T> elem` as in put or take `T` as `const T & elem` if more generic.

Reply:
agreed!